src/async-wrap.cc

```diff
@@ -181,6 +183,37 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
+  // None of the V8 calls done outside the HandleScope leak a handle. If this
+  // changes in the future then the SealHandleScope wrapping the uv_run()
+  // will catch this can cause the process to abort.
+  v8::Context::Scope context_scope(env->context());
```

Comment:
I misread the following stack:
```
FATAL ERROR: v8::HandleScope::CreateHandle() Cannot create a handle without a HandleScope
 1: node::Abort() [./node_g]
 2: 0x21986e3 [./node_g]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [./node_g]
 4: v8::Utils::ApiCheck(bool, char const*, char const*) [./node_g]
 5: v8::internal::HandleScope::Extend(v8::internal::Isolate*) [./node_g]
 6: v8::internal::HandleScope::CreateHandle(v8::internal::Isolate*, v8::internal::Object*) [./node_g]
 7: v8::internal::HandleScope::GetHandle(v8::internal::Isolate*, v8::internal::Object*) [./node_g]
 8: v8::internal::HandleBase::HandleBase(v8::internal::Object*, v8::internal::Isolate*) [./node_g]
 9: v8::internal::Handle<v8::internal::JSFunction>::Handle(v8::internal::JSFunction*, v8::internal::Isolate*) [./node_g]
10: v8::internal::Isolate::object_function() [./node_g]
11: v8::Object::New(v8::Isolate*) [./node_g]
12: node::AsyncWrap::DestroyIdsCb(uv_idle_s*) [./node_g]
```
Which is generated by the following diff when running `test-async-wrap-check-providers.js`:
```diff
diff --git a/src/async-wrap.cc b/src/async-wrap.cc
index 3887179..0bdde8d 100644
--- a/src/async-wrap.cc
+++ b/src/async-wrap.cc
@@ -190,6 +190,7 @@ void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
   // will catch this can cause the process to abort.
   v8::Context::Scope context_scope(env->context());
   Local<Function> fn = env->async_hooks_destroy_function();
+  Local<Object> test = Object::New(env->isolate());
 
   uv_idle_stop(handle);
 
```
Now I'm confused as to why the `SealHandleScope` would be needed since I get the same stack even if I add the `SealHandleScope` before it. Thoughts?

Anyway, I'll add the `HandleScope`. Hadn't yet b/c I haven't been able to write a test that would cause the script to crash w/o it. If you have an example of one that would be excellent.